src/custom/state/enhancedTransactions/updater/FinalizeTxUpdater.tsx

```diff
@@ -28,107 +31,135 @@ export function shouldCheck(lastBlockNumber: number, tx: TxInterface): boolean {
     return blocksSinceCheck > 9
   } else if (minutesPending > 5) {
     // every 3 blocks if pending more than 5 minutes
-    return blocksSinceCheck > 2
+    return blocksSinceCheck >= 3
   } else {
     // otherwise every block
     return true
   }
 }
 
-const RETRY_OPTIONS_BY_CHAIN_ID: { [chainId: number]: RetryOptions } = {
-  // [SupportedChainId.ARBITRUM_ONE]: { n: 10, minWait: 250, maxWait: 1000 },
-  // [SupportedChainId.ARBITRUM_KOVAN]: { n: 10, minWait: 250, maxWait: 1000 },
+interface CheckEthereumTransactions {
+  chainId: number
+  transactions: EnhancedTransactionDetails[]
+  lastBlockNumber: number
+  getReceipt: GetReceipt
+  dispatch: Dispatch
+  addPopup: ReturnType<typeof useAddPopup>
+}
+
+type Cancel = () => void
+
+// async function checkEthereumTransactions(transaction: EnhancedTransactionDetails) {}
+
+function finalizeEthereumTransaction(
+  receipt: TransactionReceipt,
+  transaction: EnhancedTransactionDetails,
+  params: CheckEthereumTransactions
+) {
+  const { chainId, lastBlockNumber, addPopup, dispatch } = params
+  const { hash } = transaction
+
+  dispatch(
+    finalizeTransaction({
+      chainId,
+      hash,
+      receipt: {
+        blockHash: receipt.blockHash,
+        blockNumber: receipt.blockNumber,
+        contractAddress: receipt.contractAddress,
+        from: receipt.from,
+        status: receipt.status,
+        to: receipt.to,
+        transactionHash: receipt.transactionHash,
+        transactionIndex: receipt.transactionIndex,
+      },
+    })
+  )
+
+  addPopup(
+    {
+      txn: {
+        hash,
+        success: receipt.status === 1,
+        summary: transaction.summary,
+      },
+    },
+    hash
+  )
+
+  // the receipt was fetched before the block, fast forward to that block to trigger balance updates
+  if (receipt.blockNumber > lastBlockNumber) {
+    dispatch(updateBlockNumber({ chainId, blockNumber: receipt.blockNumber }))
+  }
+}
+
+function checkEthereumTransactions(params: CheckEthereumTransactions): Cancel[] {
+  const { transactions, chainId, lastBlockNumber, getReceipt, dispatch } = params
+
+  const promiseCancellations = transactions.map((transaction) => {
+    const { hash, hashType } = transaction
+
+    if (hashType === HashType.ETHEREUM_TX) {
+      // Get receipt for transaction, and finalize if needed
+      const { promise, cancel } = getReceipt(hash)
+      promise
+        .then((receipt) => {
+          if (receipt) {
+            // If the tx is mined. We finalize it!
+            finalizeEthereumTransaction(receipt, transaction, params)
+          } else {
+            // Update the last checked blockNumber
+            dispatch(checkedTransaction({ chainId, hash, blockNumber: lastBlockNumber }))
+          }
+        })
+        .catch((error) => {
+          if (!error.isCancelledError) {
+            console.error(`[FinalizeTxUpdater] Failed to check transaction hash: ${hash}`, error)
+          }
```

Comment:
else what? no log?

Reply:
If it's canceled, no need to log anything. Shouldn't make noise. It's part of the normal flow.